### ISSUE TITLE
New pophelp: `trailing_slash`

### ISSUE THREAD

**Issue by petecooper** [textpattern/pophelp#64]
`/textpattern/index.php?event=prefs`

![Screenshot 2023-07-16 at 17 42 37](https://github.com/textpattern/pophelp/assets/226578/f468a983-712d-4a5a-8bd8-5cc5648bde2f)

- [x] create pophelp entries
- [x] add markup in https://github.com/textpattern/textpattern

**Comment by Bloke** [comment]
@philwareham Stupid question but in CrowdIn, how do I add this new string to the pophelp file? Can't for the life of me find 'sources' or any of the various 'strings' menu options that the CrowdIn docs refer to. Or is this a special case because it's an XML document? Maybe the workflow here that we make the changes to the pophelp repo directly and these are then pushed _up_ (or synced somehow) to CrowdIn?

**Comment by philwareham** [comment]
Hi Stef, Crowdin doesn't work like that. We upload a 'master' file (in this case `en_pophlep.xml`) and it uses that to provide strings to the languages in the system. However, this can be quite destructive if not uploaded properly, I've found. It can lose current translations that exist for the master. If you can update the `en` file in the Github repo and let me know when ready, I will attempt to load it into Crowdin.

**Comment by Bloke** [comment]
Cool, ta. I vaguely remember me asking this before but couldn't remember the answer. Thanks for the nudge.

I'll update just 'en' ('en-gb' and 'en-us' too? Or just 'en'?) in the Github repo.

**Comment by philwareham** [comment]
Yeah you can do, I just won't copy those back across from Crowdin.

There is quite a lot off manual labour in getting the translations from Crowdin into our repos - although _some_ of it can be automated I found that the risks outweigh the gains. For example if you move strings to a different group then the system sees those as new (blank) strings even though translations were there before - so I have to go through the files instating the translations (which luckily it saves in translation history.

I've already spent a couple of hours this week reinstating strings and coping the resulting files back to our repos, and probably only about a third of the files so far - will need to continue as I get spare moments during the week.

**Comment by Bloke** [comment]
Okay, cool. I'll do those two English ones as well.

> For example if you move strings to a different group then the system sees those as new (blank) strings even though translations were there before

Ah, balls, so my [meddling the other week](https://github.com/textpattern/textpacks/commit/46555a70e96d0bc84ed1fafab9a59972c4b25707) to remove the invalid string keys has probably fouled it up massively. Sorry about that.

If it's easier to keep the `txp_` prefixed keys in alphabetical order, so they're all lumped together, instead of it keeping them in the original order as if the prefix wasn't there, I can remove the exception in the move-strings script. Makes no odds which order they're in the file, so we should choose whichever is the path of least resistance to keep the crowdin<->strings repos->core workflow sane.

**Comment by Bloke** [comment]
@philwareham Done for en, en-us and en-gb in https://github.com/textpattern/pophelp/commit/cf227a409e4e56a7307418de50657e84ec52c74a
